### PR TITLE
Loki: Remove already selected options from next label filter options in builder

### LINKED ISSUE #57161: Loki query dashboard supports adding same label which was already chosen
### Discussed in https://github.com/grafana/grafana/discussions/53604

<div type='discussions-op-text'>

<sup>Originally posted by **SanthoshS20** August 11, 2022</sup>
**What happened:**
The label dropdown list shows the already chosen label in the query field. We can able to choose same label with same value multiple times which is not required.

**What you expected to happen:**
We should remove the labels from the dropdown list which was already chosen.

**How to reproduce it (as minimally and precisely as possible):**
**Step 1:** Go the the Loki dashboard.
**Step 2:** Switch to the builder mode.
**Step 3:** In the labels, click the + (PLUS) button to add a new label.
**Step 4:** Choose a label.
**Step 5:** Add another label by clicking the + (PLUS) button.
**Step 6:** Choose the same label which is already chosen.

**Anything else we need to know?:**
183443387-8dc58307-0c87-40ce-b981-c7242d7b4a00

**Environment:**

Grafana version: 9.0.5
User OS & Browser: Windows 10 & Chrome Browser</div>

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:

In the same way, as in code mode in autocomplete we don't offer already selected label names, we are fixing query builder to not show already selected label names.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/57161

**Special notes for your reviewer**:

To test this
1. run loki data source make `devenv sources=loki`
2. Use Loki query builder
3. Select label filter
4. Click on add label filter button
5. You should not be offered the already selected label name

### CODE REVIEW

File: public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilder.test.tsx [hunk 1]
```diff
@@ -51,6 +51,23 @@ describe('LokiQueryBuilder', () => {
...
+    const props = createDefaultProps();
+    props.datasource.getDataSamples = jest.fn().mockResolvedValue([]);
+    props.datasource.languageProvider.fetchSeriesLabels = jest
+      .fn()
+      .mockReturnValue({ job: ['a'], instance: ['b'], baz: ['bar'] });
+
+    render(<LokiQueryBuilder {...props} query={defaultQuery} />);
+    await userEvent.click(screen.getByLabelText('Add'));
+    const labels = screen.getByText(/Label filters/);
+    const selects = getAllByRole(labels.parentElement!.parentElement!.parentElement!, 'combobox');
```
Comment (matyax) [inline public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilder.test.tsx:64]: First time that I see this in a RTT test. Is this for interacting with the `select`? Maybe try `import { selectOptionInTest } from 'test/helpers/selectOptionInTest';` for that?

**Review by matyax — APPROVED** [review]
Looks good. Left a suggestion.

**Review by matyax — APPROVED** [review]
👏 👏

File: public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilder.test.tsx (hunk 1 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilder.test.tsx]: Great suggestion! Updated in https://github.com/grafana/grafana/pull/57187/commits/52f334fafe96577046e246e3ee465c8a2433d3b5

**Pull request merged**